modules/social_features/social_event/modules/social_event_an_enroll/social_event_an_enroll.module

```diff
@@ -0,0 +1,275 @@
+  $query->condition('eefe.field_event_target_id', $nid);
+
+  return (int) $query
+    ->countQuery()
+    ->execute()
+    ->fetchField();
+}
+
+/**
+ * Returns number of anonymous enrollments.
+ */
+function social_event_an_enroll_token_exists($token, $nid) {
+  $query = \Drupal::database()
+    ->select('event_enrollment__field_token', 'eeft');
+  $query->join('event_enrollment__field_event', 'eefe', 'eeft.entity_id = eefe.entity_id');
+  $query->condition('eeft.field_token_value', $token);
+  $query->condition('eefe.field_event_target_id', $nid);
+
+  $results = $query
+    ->countQuery()
+    ->execute()
+    ->fetchField();
+
+  return !empty($results);
+}
+
+/**
+ * Returns url to event with token for anonymous enrollment.
+ */
+function social_event_an_enroll_url($nid, $token) {
+  $url = Url::fromRoute('entity.node.canonical',
+    ['node' => $nid],
+    [
+      'query' => ['token' => $token],
+      'absolute' => TRUE,
+    ])->toString();
+  return $url;
+}
+
+/**
+ * Implements hook_menu_local_tasks_alter().
+ */
+function social_event_an_enroll_menu_local_tasks_alter(&$data, $route_name) {
+  $can_show_an_enrollments_link = FALSE;
+  $routes_to_check = [
+    'view.event_enrollments.view_enrollments',
+    'entity.node.canonical',
+    'view.managers.view_managers',
+    'view.manage_enrollments.page',
+  ];
+  if (in_array($route_name, $routes_to_check)) {
+    $node = \Drupal::service('current_route_match')->getParameter('node');
+    if (!is_null($node) && (!$node instanceof Node)) {
+      $node = Node::load($node);
+    }
+    if (($node instanceof Node) && $node->getType() === 'event' && $node->get('field_content_visibility')->getString() === 'public'
+    ) {
+      $can_show_an_enrollments_link = TRUE;
+    }
+  }
+  // Place this here, since hiding it should happen
+  // always and not only on the mentioned routes.
+  if (!$can_show_an_enrollments_link) {
+    unset($data['tabs'][0]['views_view:view.manage_enrollments.page']);
+  }
+}
+
+/**
+ * Check if anonymous enrollment is allowed for given event.
+ */
+function social_event_an_enroll_is_enabled(Node $node) {
+  $config = \Drupal::config('social_event_an_enroll.settings');
+  $is_global_enabled = $config->get('event_an_enroll');
+  $is_event = $node->getType() === 'event';
+  $is_public = $node->get('field_content_visibility')->getString() === 'public';
+  $is_event_an_enroll = !empty($node->get('field_event_an_enroll')->value);
+  return $is_global_enabled && $is_event && $is_public && $is_event_an_enroll;
+}
+
+/**
+ * Implements hook_form_alter().
+ */
+function social_event_an_enroll_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  $event_forms = [
+    'node_event_form',
+    'node_event_edit_form',
+  ];
+
+  if (in_array($form_id, $event_forms)) {
+    // @TODO: Fix states when group is selected.
+    $form['field_event_an_enroll']['#states'] = [
+      'visible' => [
+        ':input[name="field_content_visibility"]' => ['value' => 'public'],
```

Comment:
@ribel I placed this as a comment to potentially fix a bug that I think @jochemvn mentioned. Is that still something worth looking at?

Reply:
@Kingdutch I have tried to solve it before, and also with your suggestion, but without any luck :(
The difficulty that it's not just enough if groups were filled or not, but what exactly group was selected, becuase for public groups we need to display AN Enroll checkbox and for other - not. 
And also because Group is not a normal select field, but our custom widget, which is already altered in `social_group_field_widget_form_alter()`.